jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -83,62 +82,173 @@ impl Authz {
             )
             .set_auth_info(AuthorizationLogInfo {
-                action: request.action.clone(),
+                action: request.action,
                 context: request.context,
-                decision: decision.decision().into(),
-                principal: principal_workload_uid.to_string(),
-                diagnostics: decision.diagnostics().clone().into(),
                 resource: resource_uid.to_string(),
+
+                person_principal: principal_user_entity_uid.to_string(),
+                workload_principal: principal_workload_uid.to_string(),
+
+                person_diagnostics: person_result.diagnostics().clone().into(),
+                workload_diagnostics: workload_result.diagnostics().clone().into(),
+
+                person_decision: person_result.decision().into(),
+                workload_decision: workload_result.decision().into(),
             })
-            .set_message("Result of authorize with resource as workload entity".to_string()),
+            .set_message("Result of authorize.".to_string()),
+        );
+
+        Ok(AuthorizeResult {
+            workload: workload_result,
+            person: person_result,
+        })
+    }
+
+    /// Execute cedar policy is_authorized method to check
+    /// if allowed make request with given parameters
+    fn execute_authorize(
+        &self,
+        parameters: ExecuteAuthorizeParameters,
+    ) -> Result<cedar_policy::Response, cedar_policy::RequestValidationError> {
+        let request_principal_workload = cedar_policy::Request::new(
+            parameters.principal,
+            parameters.action,
+            parameters.resource,
+            parameters.context,
+            Some(&self.policy_store.schema.schema),
+        )?;
+
+        let response = self.authorizer.is_authorized(
+            &request_principal_workload,
+            &self.policy_store.policies,
+            parameters.entities,
         );
 
-        Ok(AuthorizeResult { workload: decision })
+        Ok(response)
+    }
+
+    /// Create all [`Entity`]-s from [`Request`]
+    fn authorize_entities_data(
+        &self,
+        request: &Request,
+    ) -> Result<AuthorizeEntitiesData, AuthorizeError> {
+        let decoded_tokens = decode_tokens(&request, self.jwt_service.as_ref())?;
+
+        // Using the builder at compile time ensures all entities are added correctly
+        let data = AuthorizeEntitiesData::builder()
+            .access_token_entities(create_access_token_entities(
+                &self.policy_store.schema.json,
+                &decoded_tokens.access_token,
+            )?)
+            .id_token_entity(
+                id_token_entity(&self.policy_store.schema.json, &decoded_tokens.id_token)
+                    .map_err(AuthorizeError::CreateIdTokenEntity)?,
+            )
+            .user_entity(
+                user_entity(
+                    &self.policy_store.schema.json,
+                    &decoded_tokens.id_token,
```

Comment:
To avoid the cloning in `token_data.rs`
```suggestion
                    &mut decoded_tokens.id_token,
```

Reply:
`Value object` should stay immutable.